tests/components/inels/test_config_flow.py

```diff
@@ -0,0 +1,384 @@
+"""Test config flow."""
+from unittest.mock import patch
+
+from inelsmqtt import InelsMqtt
+from inelsmqtt.const import MQTT_TRANSPORT
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components import inels
+from homeassistant.components.hassio import HassioServiceInfo
+from homeassistant.components.inels.config_flow import (
+    connect_val_to_error,
+    try_connection,
+)
+from homeassistant.const import (
+    CONF_DISCOVERY,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_PROTOCOL,
+    CONF_SSL,
+    CONF_TYPE,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=True)
+def mock_is_available():
+    """Mock test_connection inside InelsMqtt."""
+    with patch(
+        "inelsmqtt.InelsMqtt.test_connection", return_value=True
```

Comment:
You should mock locally within the component whenever possible instead of globally

Reply:
@Lash-L  In that particular case, I create an instance of the class to test the connection. I think this is correct in this case, but feel free to correct me.